lib/resource-providers/kms-key.ts

```diff
@@ -15,22 +15,29 @@ import { ResourceContext, ResourceProvider } from "../spi";
  */
 export class CreateKmsKeyProvider implements ResourceProvider<kms.IKey> {
   private readonly aliasName?: string;
+  private readonly keyId?: string;
   private readonly kmsKeyProps?: kms.KeyProps;
 
   /**
    * Configuration options for the KMS Key.
    *
    * @param aliasName The alias name for the KMS Key
+   * @param keyId The key id for the KMS Key
    * @param kmsKeyProps The key props used
    */
-  public constructor(aliasName?: string, kmsKeyProps?: kms.KeyProps) {
+  public constructor(
+    aliasName?: string,
+    keyId?: string,
+    kmsKeyProps?: kms.KeyProps
+  ) {
     this.aliasName = aliasName;
+    this.keyId = keyId;
     this.kmsKeyProps = kmsKeyProps;
   }
 
   provide(context: ResourceContext): kms.IKey {
     const id = context.scope.node.id;
-    const keyId = `${id}-${this.aliasName ?? "default"}-KmsKey`;
+    const keyId = this.keyId || `${id}-${this.aliasName ?? "default"}-KmsKey`;
```

Comment:
Why not make it identical to the way it was in 1.6.0 [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/blueprints-1.6.0/lib/resource-providers/kms-key.ts)?

this will not solve the backwards compatibility issue others. 

That means if id is not pass, it must be set to:
```
`${id}-kms-key`
```

Reply:
[In this case](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/blueprints-1.6.0/lib/resource-providers/kms-key.ts#L42) you can create only one KMS key per stack. E.g., you cannot create a separate key for EBS encryption-at-rest and for EFS encryption-at-rest.

```
    const ebsKmsKeyName = "ebs-kms-encryption-key";
    const ebsKmsKey = blueprints.getNamedResource(ebsKmsKeyName) as kms.Key;

    const efsKmsKeyName = "efs-kms-encryption-key";
    const efsKmsKey = blueprints.getNamedResource(efsKmsKeyName) as kms.Key;

    EksBlueprint.builder()
      .resourceProvider(GlobalResources.Vpc, new blueprints.VpcProvider())
      // create KMS keys
      .resourceProvider(
        GlobalResources.KmsKey,
        new blueprints.CreateKmsKeyProvider()
      )
      .resourceProvider(
        ebsKmsKeyName,
        new blueprints.CreateKmsKeyProvider(ebsKmsKeyName)
      )
      .resourceProvider(
        efsKmsKeyName,
        new blueprints.CreateKmsKeyProvider(efsKmsKeyName)
      )
```

Reply:
@shapirov103 got it, thank you for the explanation. Please check https://github.com/aws-quickstart/cdk-eks-blueprints/pull/646/commits/9fae44b7793529fe6737d70c57a557dd4ccd201a

Reply:
@aliaksei-ivanou you can however, implement your own key provider (I was already doing this before). It's only a matter of a couple LoCs:
```ts
EksBlueprint.builder()
.resourceProvider(ebsKmsKeyName, {
  provide(context: ResourceContext) {
    return new kms.Key(context.scope, 'ebs-csi-encryption-key', {
      description: 'Key used by EBS CSI driver in EKS cluster ' + clusterName
    })
  }
})
```